rasa/shared/nlu/training_data/features.py

```diff
@@ -35,6 +38,25 @@ def __init__(
                 "features or a scipy sparse matrix for sparse features."
             )
 
+    def __repr__(self) -> Text:
+        return (
+            f"{self.__class__.__name__}("
+            f"features={self.features}, "
+            f"type={self.type}, "
+            f"origin={self.origin}, "
+            f"attribute={self.attribute})"
+        )
+
+    def __str__(self) -> Text:
+        return (
+            f"{self.__class__.__name__}("
+            f"features.shape={self.features.shape}, "
+            f"is_sparse={self.is_sparse()}, "
+            f"type={self.type}, "
+            f"origin={self.origin}, "
+            f"attribute={self.attribute})"
+        )
```

Comment:
`repr` prints the complete `.features` (`np.ndarray`) while `str` just prints the shape and whether it is sparse -- think both might be useful for debugging (though I mostly used the latter) and it might be better that repr prints the complete thing (because ideally one should be able to re-create the object with that output, no?)

Reply:
ups, haven't seen the difference. Makes sense 👍🏻